v2/pkg/engine/resolve/resolve.go

```diff
@@ -296,6 +296,45 @@ type sub struct {
 	executor chan func()
 }
 
+func (r *Resolver) ResolveGraphQLIncrementalResponse(ctx *Context, response *GraphQLIncrementalResponse, writer io.Writer) error {
```

Comment:
I'd create an interface as writer, e.g. IncrementalResponseWriter with 3 funcs, write, flush, complete. This way, you don't have to think about boundaries or transports here. Just write a JSON and flush it. Let the implementer of the writer adapt the transport, e.g. by adding a boundary after flush.